gadget/install/encrypt.go

```diff
@@ -103,3 +106,73 @@ func cryptsetupClose(name string) error {
 	}
 	return nil
 }
+
+// encryptedDeviceWithSetupHook represents a block device that is setup using
+// the "device-setup" hook.
+type encryptedDeviceWithSetupHook struct {
+	parent *gadget.OnDiskStructure
+	name   string
+	node   string
+}
+
+// sanity
+var _ = encryptedDevice(&encryptedDeviceWithSetupHook{})
+
+// newEncryptedDeviceWithSetupHook creates an encrypted device in the
+// existing partition using the specified key using the fde-setup hook
+func newEncryptedDeviceWithSetupHook(part *gadget.OnDiskStructure, key secboot.EncryptionKey, name string) (encryptedDevice, error) {
```

Comment:
sorry for my usual naming nitpicks :-), but when I see `new` I assume that the function creates a new object in memory; given the doc comment mentions `creates` and that this function invokes `CreateLinearMapperDevice()`, I think that the `create` prefix would be more appropriate here.

Reply:
done